generators/entity/templates/src/main/java/package/domain/_Entity.java

```diff
@@ -154,6 +159,9 @@
     @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
     <%_     }
             if (ownerSide == true) { _%>
+    <%_ if (relationshipRequired) { _%>
```

Comment:
better check `<%_ if (relationshipValidate ) { _%>` alone here to future proof so that other validations can also be added if required, you are already checking the required condition in the ejs file so thats sufficient